File: src/main/java/main/ui/tip/TipMouseListener.java
package main.ui.tip;

import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import java.awt.*;
import java.awt.event.MouseEvent;
import java.awt.event.MouseListener;
import java.util.HashMap;
import java.util.Map;

/**
 *
 * @author Bunnyspa
 */
public class TipMouseListener implements MouseListener {

    private final JLabel label;
    @NotNull
    private final Map<Component, String> map;

    public TipMouseListener(JLabel label) {
        map = new HashMap<>();
        this.label = label;
    }

    public void clearTips() {
        map.clear();
    }

    public void setTip(Component c, String s) {
        map.put(c, s);
    }

    @Override
    public void mouseClicked(MouseEvent e) {
    }

    @Override
    public void mousePressed(MouseEvent e) {
    }

    @Override
    public void mouseReleased(MouseEvent e) {
    }

    @Override
    public void mouseEntered(@NotNull MouseEvent e) {
        Component c = e.getComponent();
        if (c != null && map.containsKey(c)) {
            label.setText(map.get(c));
        } else {
            label.setText(" ");
        }
    }

    @Override
    public void mouseExited(MouseEvent e) {
        label.setText(" ");
    }

}

File: src/main/java/main/ui/tip/TipMouseListener.kt
package main.ui.tip

import java.awt.Component
import java.awt.event.MouseEvent
import java.awt.event.MouseListener
import java.util.*
import javax.swing.JLabel

/**
 *
 * @author Bunnyspa
 */
class TipMouseListener(label: JLabel?) : MouseListener {
    private val label: JLabel?
    private val map: MutableMap<Component, String?>
    fun clearTips() {
        map.clear()
    }

    fun setTip(c: Component, s: String?) {
        map[c] = s
    }

    override fun mouseClicked(e: MouseEvent) {}
    override fun mousePressed(e: MouseEvent) {}
    override fun mouseReleased(e: MouseEvent) {}
    override fun mouseEntered(e: MouseEvent) {
        val c = e.component
        if (c != null && map.containsKey(c)) {
            label!!.text = map[c]
        } else {
            label!!.text = " "
        }
    }

    override fun mouseExited(e: MouseEvent) {
        label!!.text = " "
    }

    init {
        map = HashMap()
        this.label = label
    }
}